test/github-issues/4429/issue-4429.ts

```diff
@@ -0,0 +1,43 @@
+import "reflect-metadata";
+import {Connection} from "../../../src/connection/Connection";
+import {closeTestingConnections, createTestingConnections} from "../../utils/test-utils";
+import { Category } from "./entity/Category";
+import { Post } from "./entity/Post";
+import { expect } from "chai";
+import { LessThan } from "../../../src/find-options/operator/LessThan";
+
+describe("github issues > #4429 Incorrect construction of query with innerJoin and where ", () => {
+
+    let connections: Connection[];
+    before(async () => {
+        connections = await createTestingConnections({
+            entities: [__dirname + "/entity/*{.js,.ts}"],
+            enabledDrivers: ["mysql"],
```

Comment:
Can this be enabled for other drivers

Reply:
Feel free to push into the `mixed-placeholder` branch of my fork, I allowed pushes for maintainers. Or cherry-pick my commit into a branch of your own.

I only use MySQL myself, thus I don't feel confident making this change.

Thanks.